03_oop_part_2/src/pages.ts

```diff
@@ -0,0 +1,24 @@
+import { Page } from './page';
+
+export class Pages implements IterableIterator<Page> {
+
+    private pointer = 0;
+
+    constructor(private pages: Page[]) {}
+
+    public next(): IteratorResult<Page> {
+        if (this.pointer < this.pages.length) {
+            return {done: false, value: this.pages[this.pointer++]};
+        }
+        this.pointer = 0;
+        return {done: true, value: null};
+    }
+
+    [Symbol.iterator](): IterableIterator<Page> {
```

Comment:
`Pages` class itself shouldn't be iterable. Iteration logic shoud reside in `PagesIterable` mixin